llvm/lib/Target/AMDGPU/SIPeepholeSDWA.cpp

```diff
@@ -327,7 +336,37 @@ uint64_t SDWASrcOperand::getSrcMods(const SIInstrInfo *TII,
   return Mods;
 }
 
-MachineInstr *SDWASrcOperand::potentialToConvert(const SIInstrInfo *TII) {
+MachineInstr *SDWASrcOperand::potentialToConvert(const SIInstrInfo *TII,
+                                                 const GCNSubtarget &ST,
+                                                 SDWAOperandsMap *PotentialMatches) {
+  // If PotentialMatches is not null, then fill out the map for all uses,
+  // if all can be converted
+  if (PotentialMatches != nullptr) {
+    MachineOperand *Reg = getReplacedOperand();
+    if (!Reg->isReg() || !Reg->isDef()) {
+      return nullptr;
+    }
+
+    for (MachineOperand &UseMO : getMRI()->use_nodbg_operands(Reg->getReg())) {
```

Comment:
use instructions instead of operands? You do the full instruction legality check

Reply:
I iterated on operands because of the subreg issue. Otherwise we may inadvertently mark an operand as foldable. I do observe that we should never get one, though, as "matchSDWAOperands" shouldn't allow it to happen. I can add an assert into the next loop and convert this to instructions.